test/policy-collection/policy_generator_acm_hardening.go

```diff
@@ -0,0 +1,256 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/stolostron/governance-policy-framework/test/common"
+)
+
+// cleanup will remove any test data/configuration on the OpenShift cluster that was added/updated
+// as part of the policy generator test. Any errors will be propagated as gomega failed assertions.
+func cleanup(namespace string, secret string, user common.OCPUser) {
+	err := clientHub.CoreV1().Namespaces().Delete(context.TODO(), namespace, metav1.DeleteOptions{})
+	if !k8serrors.IsNotFound(err) {
+		Expect(err).Should(BeNil())
+	}
+
+	// Wait for the namespace to be fully deleted before proceeding.
+	Eventually(
+		func() bool {
+			_, err := clientHub.CoreV1().Namespaces().Get(
+				context.TODO(), namespace, metav1.GetOptions{},
+			)
+			return k8serrors.IsNotFound(err)
+		},
+		defaultTimeoutSeconds,
+		1,
+	).Should(BeTrue())
+
+	err = common.CleanupOCPUser(clientHub, clientHubDynamic, secret, user)
+	Expect(err).Should(BeNil())
+
+	err = clientHub.CoreV1().Secrets("openshift-config").Delete(context.TODO(), secret, metav1.DeleteOptions{})
+	if !k8serrors.IsNotFound(err) {
+		Expect(err).Should(BeNil())
+	}
+}
+
+var _ = Describe("GRC: [P1][Sev1][policy-grc] Test the ACM Hardening generated PolicySet in an App subscription", func() {
+	const namespace = "policies"
+	const secret = "grc-e2e-subscription-admin-user"
+	const clustersetRoleName = "grc-e2e-clusterset-role"
+	const subAdminBinding = "open-cluster-management:subscription-admin"
+	ocpUser := common.OCPUser{
+		ClusterRoles: []types.NamespacedName{
+			{Name: "open-cluster-management:admin:local-cluster"},
+			{
+				Name:      "admin",
+				Namespace: namespace,
+			},
+			{Name: "clustersetRoleName"},
```

Comment:
Did you mean this?
```suggestion
			{Name: clustersetRoleName},
```

Reply:
I wonder if I need to test on a clean system :-)